torch/distributed/_composable/fully_shard.py

```diff
@@ -98,7 +98,8 @@ def fully_shard(
     for submodule in module.modules():
         if (
             submodule not in state._ignored_modules
-            and _get_module_state(submodule) is None
+            and _get_module_state(submodule)
+            is not state  # either `None`or different state
```

Comment:
- `state` is not `None`.
- `_get_module_state(submodule)` returns `None` or some `_state`.
- Since we do not support manual "wrapping" yet, `_get_module_state(submodule)` must return `None` or the same `state`. (I.e., `_state is state`.) However, after we support manual "wrapping", it may be that `_state is not state`.